### PR TITLE
SD-1357 move core create & update address services to frontend addresses controller

### PR DESCRIPTION
**Pull request by KacperMekarski**

### CODE REVIEW

File: frontend/app/controllers/spree/addresses_controller.rb [hunk 1]
```diff
@@ -60,5 +47,13 @@ def destroy
     def address_params
       params.require(:address).permit(permitted_address_attributes)
     end
+
+    def create_service
+      Spree::Api::Dependencies.storefront_account_create_address_service.constantize
```
Comment (damianlegawiec) [inline frontend/app/controllers/spree/addresses_controller.rb:52]:
```suggestion
      Spree::Dependencies.account_create_address_service.constantize
```

File: frontend/app/controllers/spree/addresses_controller.rb [hunk 2]
```diff
@@ -60,5 +47,13 @@ def destroy
...
+    end
+
+    def update_service
+      Spree::Api::Dependencies.storefront_account_update_address_service.constantize
```
Comment (damianlegawiec) [inline frontend/app/controllers/spree/addresses_controller.rb:56]: @KacperMekarski and similarly here

File: frontend/app/controllers/spree/addresses_controller.rb [hunk 3]
```diff
@@ -60,5 +47,15 @@ def destroy
     def address_params
       params.require(:address).permit(permitted_address_attributes)
     end
+
+    def create_service
+      binding.pry
```
Comment (damianlegawiec) [inline frontend/app/controllers/spree/addresses_controller.rb:52]: @KacperMekarski pry alert :)

File: frontend/app/controllers/spree/addresses_controller.rb [hunk 4]
```diff
@@ -60,5 +47,15 @@ def destroy
...
+      Spree::Dependencies.account_create_address_service.constantize
+    end
+
+    def update_service
+      binding.pry
```
Comment (damianlegawiec) [inline frontend/app/controllers/spree/addresses_controller.rb:57]: and here

File: frontend/app/controllers/spree/addresses_controller.rb [hunk 5]
```diff
@@ -10,7 +10,7 @@ def index
 
     def create
       @address = try_spree_current_user.addresses.build(address_params)
-      if @address.save
+      if create_service.call(user: try_spree_current_user, address_params: @address.attributes).success?
         flash[:notice] = I18n.t(:successfully_created, scope: :address_book)
         redirect_to spree.account_path
       else
```
Comment (damianlegawiec) [inline frontend/app/controllers/spree/addresses_controller.rb:16]: @KacperMekarski shouldn't we pass the result to the view? will it still display validation errors?

Comment (KacperMekarski) [reply frontend/app/controllers/spree/addresses_controller.rb:16]: @damianlegawiec views are using @address variable which did not change.
I checked it locally, it works fine :)
Same thing happens with #update.

**Review by damianlegawiec — APPROVED** [review]

File: frontend/spec/controllers/spree/addresses_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+          expect{ put_update }.to change { Spree::Address.count }.by(1)
+        end
+
+        it 'sets deleted_at attribute of original address' do
+          Timecop.freeze(Time.current) do
+            expect(address.deleted_at).to be_nil
+
+            put_update
+
+            expect(address.deleted_at).to eq Time.current
```
Comment (damianlegawiec) [inline frontend/spec/controllers/spree/addresses_controller_spec.rb:147]:
```suggestion
            expect(address.deleted_at).not_to be nil
```

checking `Time.current` will always be buggy, even with timecop

**Pull request merged**